### PR TITLE
Made enhancements to autolog the html representation of the estimators

### LINKED ISSUE #6758: [FR] Log HTML representation of estimator in scikit-learn autologging
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

Log an HTML representation of an estimator in scikit-learn autologging using [`sklearn.utils.estimator_html_repr`](https://scikit-learn.org/stable/modules/generated/sklearn.utils.estimator_html_repr.html#sklearn.utils.estimator_html_repr).

image

### Motivation

> #### What is the use case for this feature?

To make it easier to understand the structure of a auto-logged estimator.

> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

### Details

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by RohanBha1**
## Related Issues/PRs

Close #6758 

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

MLflow users now can view quickly view estimator visualizations on the MLflow UI.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@mlflow-automation autoforamt

File: mlflow/sklearn/utils.py [hunk 1]
```diff
@@ -553,6 +553,24 @@ def _log_specialized_estimator_content(
...
+    <html lang = "en">
+        <head>
+            <meta charset = "UTF-8"/>
+        </head>
+        <body>
+            {estimator_html_repr.estimator_html_repr(estimator)}
+        </body>
+    </html>
+    """
+    MlflowClient.log_text(run_id, estimator_html_string, artifact_file="estimator.html")
```
Comment (harupy) [inline mlflow/sklearn/utils.py:571]:
```suggestion
    MlflowClient().log_text(run_id, estimator_html_string, artifact_file="estimator.html")
```

File: mlflow/sklearn/utils.py [hunk 2]
```diff
@@ -553,6 +553,24 @@ def _log_specialized_estimator_content(
...
+
+    # Specifies charset so triangle toggle buttons are not garbled
+    estimator_html_string = f"""
+    <!DOCTYPE html>
+    <html lang = "en">
+        <head>
+            <meta charset = "UTF-8"/>
+        </head>
+        <body>
+            {estimator_html_repr.estimator_html_repr(estimator)}
```
Comment (harupy) [inline mlflow/sklearn/utils.py:567]:
```suggestion
            {estimator_html_repr(estimator)}
```

File: mlflow/sklearn/utils.py [hunk 3]
```diff
@@ -553,6 +553,24 @@ def _log_specialized_estimator_content(
...
+    estimator_html_string = f"""
+    <!DOCTYPE html>
+    <html lang = "en">
+        <head>
+            <meta charset = "UTF-8"/>
+        </head>
+        <body>
+            {estimator_html_repr(estimator)}
+        </body>
+    </html>
```
Comment (harupy) [inline mlflow/sklearn/utils.py:569]:
```suggestion
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8"/>
  </head>
  <body>
    {estimator_html_repr(estimator)}
  </body>
</html>
```

**Comment by harupy** [comment]
@RohanBha1 Can you fix the failed tests?

**Comment by RohanBha1** [comment]
@harupy what else would I need to change?

**Comment by harupy** [comment]
@RohanBha1 We need to fix the failed tests. Can I push a commit?

**Comment by RohanBha1** [comment]
@harupy go ahead? How do you know where the failed test cases are?

**Comment by harupy** [comment]
@RohanBha1 You can click the red cross icons in https://github.com/mlflow/mlflow/pull/6956/commits to see the failed CI checks.

**Comment by harupy** [comment]
@RohanBha1 Pushed two commits to fix failed tests.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**